sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -106,6 +108,22 @@ object TypeCoercion {
     case (t1, t2) => findTypeForComplex(t1, t2, findTightestCommonType)
   }
 
+  /**
+   * Finds a wider decimal type between the two supplied decimal types without
+   * any loss of precision.
+   */
+  def findWiderDecimalType(d1: DecimalType, d2: DecimalType): Option[DecimalType] = {
```

Comment:
Is there a reference for this implementation? I'm worried about corner cases like negative scale.

Reply:
@cloud-fan Actually the bounded version is in `DecimalPrecision::widerDecimalType`. Thats the function i looked at as reference.

Reply:
ok. Then can we add some more tests with negative scale?

Reply:
@cloud-fan Added tests with -ve scale. Thanks !!